[scroll-animations] WPT test `scroll-animations/css/view-timeline-inset-animation.html` is a failure

graouts opened the pull request:
#### bf12caa476dbd8ee4dfb76b1b4568f874fe85a30
<pre>
[scroll-animations] WPT test `scroll-animations/css/view-timeline-inset-animation.html` is a failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=287790">https://bugs.webkit.org/show_bug.cgi?id=287790</a>

Reviewed by Anne van Kesteren.

We were failing the test `scroll-animations/css/view-timeline-inset-animation.html` because of how we handle `margin`
values to determine the offset width and height of the scrollable content. But the expected behavior is not strictly
specified by the css-overflow spec, per <a href="https://drafts.csswg.org/css-overflow-3/#scrollable">https://drafts.csswg.org/css-overflow-3/#scrollable</a>:

&gt; The UA may additionally include the margin areas of other boxes for which the box establishes
&gt; a containing block; however, the conditions under which such margin areas are included is
&gt; undefined in this level.

Additionally, the use of `margin` itself is not directly relevant to the test, but was just a test authoring decision
to easily place the view timeline subject in the center of a scrollable region. So we rewrite this WPT test to use
absolute positioning for the subject and explicit scrollable content sizing with an intermediate `#container` element.

Note that bringing our `margin` handling in line with Chrome and Firefox for interoperability reasons is covered by
<a href="https://bugs.webkit.org/show_bug.cgi?id=287802.">https://bugs.webkit.org/show_bug.cgi?id=287802.</a> A new WPT test is being added to `css/css-overflow` to cover such cases
so that we don&apos;t lose test coverage.

* LayoutTests/imported/w3c/web-platform-tests/css/css-overflow/scrollable-overflow-height-with-flex-item-margin-inline-end-rtl-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-overflow/scrollable-overflow-height-with-flex-item-margin-inline-end-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-inset-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-inset-animation.html:

Canonical link: <a href="https://commits.webkit.org/290603@main">https://commits.webkit.org/290603@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c6bc4a5d8726ef6e795e3501eaf4382c316b1786

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/90589 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/10121 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/45527 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/95601 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/41371 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/10518 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/18440 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69706 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27261 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/93590 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/8018 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/82154 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50058 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7766 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/36527 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/40501 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/78074 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/37593 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/97430 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/17781 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/13069 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/78732 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18039 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/77979 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77932 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22374 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/20993 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/11103 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14255 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/17791 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/17530 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/20985 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/19314 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (commented):
I would prefer it if you also checked in a test that Safari continues to fail specifically for the margin issue. That way we don't regress overall coverage.

graouts committed:
[scroll-animations] WPT test `scroll-animations/css/view-timeline-inset-animation.html` is a failure
https://bugs.webkit.org/show_bug.cgi?id=287790

Reviewed by Anne van Kesteren.

We were failing the test `scroll-animations/css/view-timeline-inset-animation.html` because of how we handle `margin`
values to determine the offset width and height of the scrollable content. But the expected behavior is not strictly
specified by the css-overflow spec, per https://drafts.csswg.org/css-overflow-3/#scrollable:

> The UA may additionally include the margin areas of other boxes for which the box establishes
> a containing block; however, the conditions under which such margin areas are included is
> undefined in this level.

Additionally, the use of `margin` itself is not directly relevant to the test, but was just a test authoring decision
to easily place the view timeline subject in the center of a scrollable region. So we rewrite this WPT test to use
absolute positioning for the subject and explicit scrollable content sizing with an intermediate `#container` element.

Note that bringing our `margin` handling in line with Chrome and Firefox for interoperability reasons is covered by
https://bugs.webkit.org/show_bug.cgi?id=287802. A new WPT test is being added to `css/css-overflow` to cover such cases
so that we don't lose test coverage.

* LayoutTests/imported/w3c/web-platform-tests/css/css-overflow/scrollable-overflow-height-with-flex-item-margin-inline-end-rtl-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-overflow/scrollable-overflow-height-with-flex-item-margin-inline-end-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-inset-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-inset-animation.html:

Canonical link: https://commits.webkit.org/290603@main
